bin/generate-php-sync-issue.mjs

```diff
@@ -0,0 +1,471 @@
+	return octokitRequest( method, params, { paginate: true } );
+}
+
+async function octokitRequest( method = '', params = {}, settings = {} ) {
+	const octokit = new Octokit( { auth: authToken } );
+	params.owner = OWNER;
+	params.repo = REPO;
+
+	const requestType = settings?.paginate ? 'paginate' : 'request';
+
+	try {
+		const result = await octokit[ requestType ]( method, params );
+
+		if ( requestType === 'paginate' ) {
+			return result;
+		}
+		return result.data;
+	} catch ( error ) {
+		console.error(
+			`Error making request to ${ method }: ${ error.message }`
+		);
+		process.exit( 1 );
+	}
+}
+
+async function fetchAllCommitsFromPaths( since, paths ) {
+	let commits = [];
+
+	for ( const path of paths ) {
+		const pathCommits = await fetchAllCommits( since, path );
+		commits = [ ...commits, ...pathCommits ];
+	}
+
+	return commits;
+}
+
+function buildPRURL( commit ) {
+	const prIdMatch = commit.commit.message.match( /\(#(\d+)\)/ );
+	const prId = prIdMatch ? prIdMatch[ 1 ] : null;
+	return prId
+		? `https://github.com/WordPress/gutenberg/pull/${ prId }`
+		: `[Commit](${ commit.html_url })`;
+}
+
+function sortLevels( data ) {
+	function processLevel( levelData ) {
+		const processedData = {};
+
+		// Separate directories and files
+		const directories = {};
+		const files = {};
+
+		for ( const [ key, value ] of Object.entries( levelData ) ) {
+			if ( key.endsWith( '.php' ) ) {
+				files[ key ] = Array.isArray( value )
+					? value
+					: processLevel( value );
+			} else {
+				directories[ key ] = Array.isArray( value )
+					? value
+					: processLevel( value );
+			}
+		}
+
+		// Combine directories and files
+		Object.assign( processedData, directories, files );
+
+		return processedData;
+	}
+
+	return processLevel( data );
+}
+
+function removeEmptyLevels( data ) {
+	function processLevel( levelData ) {
+		const processedData = {};
+
+		for ( const [ key, value ] of Object.entries( levelData ) ) {
+			if ( Array.isArray( value ) ) {
+				if ( value.length > 0 ) {
+					processedData[ key ] = value;
+				}
+			} else {
+				const processedLevel = processLevel( value );
+				if ( Object.keys( processedLevel ).length > 0 ) {
+					processedData[ key ] = processedLevel;
+				}
+			}
+		}
+
+		return processedData;
+	}
+
+	return processLevel( data );
+}
+
+function dedupePRsPerLevel( data ) {
+	function processLevel( levelData ) {
+		const processedData = {};
+		const prSet = new Set();
+
+		for ( const [ key, value ] of Object.entries( levelData ) ) {
+			if ( Array.isArray( value ) ) {
+				processedData[ key ] = value.filter( ( commit ) => {
+					if ( ! prSet.has( commit.pullRequest.url ) ) {
+						prSet.add( commit.pullRequest.url );
+						return true;
+					}
+					return false;
+				} );
+			} else {
+				processedData[ key ] = processLevel( value );
+			}
+		}
+
+		return processedData;
+	}
+
+	return processLevel( data );
+}
+
+function removeSinglePRLevels( data ) {
```

Comment:
There is a bug here where the individual commits are being removed from the data rather than moved to their parent.

Reply:
I decided to remove this for now. If we want to restore the feature to remove single-PR levels then we'll probably need to adjust the data structure to be:

```
{
'lib': {
commits: [], // any commits for this level
children: {} // more leels here
}
```

That will afford a means to move any PRs listed under a level which only has a single commit listed to instead be placed on the parent level's commits.